html/browsers/browsing-the-web/navigating-across-documents/initial-empty-document/resources/helpers.js

```diff
@@ -0,0 +1,99 @@
+const url_204 = "/common/blank.html?pipe=status(204)";
+
+window.waitForLoad = (t, iframe, urlRelativeToThisDocument) => {
+  return new Promise(resolve => {
+    iframe.addEventListener("load", t.step_func(() => {
+      assert_equals(iframe.contentWindow.location.href, (new URL(urlRelativeToThisDocument, location.href)).href);
+
+      // Wait a bit longer to ensure all history stuff has settled, e.g. the document is "completely loaded"
+      // (which happens from a queued task).
+      setTimeout(resolve, 0);
+    }), { once: true });
+  });
+};
+
+window.waitForMessage = (t, message) => {
+  return new Promise(resolve => {
+    window.addEventListener("message", t.step_func((event) => {
+      if (event.data == message)
+        resolve();
+    }), { once: true });
+  });
+};
+
+window.insertIframe = (t) => {
+  const iframe = document.createElement("iframe");
+  t.add_cleanup(() => iframe.remove());
+  document.body.append(iframe);
+  return iframe;
+};
+
+window.windowOpenNoURL = (t) => {
+  const openedWindow = window.open();
+  t.add_cleanup(() => openedWindow.close());
+  return openedWindow;
+};
+
+window.insertIframeWith204Src = (t) => {
+  const iframe = document.createElement("iframe");
+  iframe.src = url_204;
+  t.add_cleanup(() => iframe.remove());
+  document.body.append(iframe);
+  return iframe;
+};
+
+window.windowOpen204 = (t) => {
+  const openedWindow = window.open(url_204);
+  t.add_cleanup(() => openedWindow.close());
+  return openedWindow;
+};
+
+window.insertIframeWithAboutBlankSrc = (t) => {
+  const iframe = document.createElement("iframe");
+  t.add_cleanup(() => iframe.remove());
+  iframe.src = "about:blank";
+  document.body.append(iframe);
+  return iframe;
+};
+
+window.insertIframeWithAboutBlankSrcWaitForLoad = async (t) => {
+  const iframe = insertIframeWithAboutBlankSrc(t);
+  const aboutBlankLoad = new Promise(resolve => {
+    if (iframe.contentDocument.readyState === 'complete') {
+      // Wait a bit longer in case the about:blank navigation after the
+      // initial empty document is asynchronous.
+      setTimeout(resolve, 100);
```

Comment:
In general all uses of `setTimeout` should instead be `t.step_timeout`, which gives them extra time on slow test runners.

Reply:
Thanks, changed!